x-pack/solutions/observability/plugins/inventory/server/routes/entity_definition/get_entity_definitions.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { createInventoryServerRoute } from '../create_inventory_server_route';
+
+export const getEntityDefinitionSourceIndexPatternsByType = createInventoryServerRoute({
+  endpoint: 'GET /internal/inventory/entity/definitions/sources',
+  options: {
+    tags: ['access:inventory'],
+  },
+  async handler({ context, request, plugins }) {
+    const [_coreContext, entityManagerStart] = await Promise.all([
+      context.core,
+      plugins.entityManager.start(),
+    ]);
+    const entityManagerClient = await entityManagerStart.getScopedClient({ request });
+
+    const entityDefinitionsSource = await entityManagerClient.v2.readSourceDefinitions({});
+
+    const allEntityDefinitionIndexPatterns = entityDefinitionsSource.map((source) => ({
+      [source.type_id]: [
+        ...new Set(
+          entityDefinitionsSource
+            .filter((sourceToFilter) => sourceToFilter.type_id === source.type_id)
+            .flatMap((filteredSource) => filteredSource.index_patterns)
+        ),
+      ],
+    }));
+
+    return {
+      definitionIndexPatterns: {
+        ...Object.fromEntries(
+          Array.from(new Set(allEntityDefinitionIndexPatterns)).flatMap(Object.entries)
```

Comment:
Is this necessary? Since the array is already structured in line 24, couldn't be formatted there?

Reply:
I don't think I understand the issue. The idea of this is to format the data in a format where all the unique values are extracted and converted to an object like:  { `unique type name`: [ all unique array values (without duplicates)] } - the mapping will return an array were here we want to get the unique key/values. 

For context: we can have more than one source per type ([comment](https://github.com/elastic/kibana/pull/204026#discussion_r1910270343)) so from the sources we can have for example: type `n` with 2 different index pattern arrays:  [index1], [index2] ... and [indexN] and we want to transform [{n: [index1]},....] to {n: [index1, index2....]} - that's why we need the set conversion/flatmap conversion of the `allEntityDefinitionIndexPatterns`

Does it make sense now?

Reply:
Update: Based on the suggestions I see that that wasn't super clear so I refactored that part and added a test for it in this commit https://github.com/elastic/kibana/pull/204026/commits/68d076c53ec315800c3f23c15f2b1bbbca33904e